### ISSUE TITLE
ADALM2000: Segfault immediately after attempting to calibrate

### ISSUE THREAD

**Issue by butterflyhigh** [analogdevicesinc/scopy#1771]
**Environment:**
 - OS: 6.10.10-arch1-1 (Linux)
 - FW: 0.27
 - SW: 1.4.1

**Describe the bug**
When I attempt to calibrate the ADALM2000 by clicking the calibrate button, Scopy crashes immediately. If started from the terminal (via flatpak run), I get this output:

```
QIODevice::read (QFile, ":/stylesheets/themes/global.qss"): device not open
QIODevice::read (QFile, ":stylesheets/stylesheets/browserStylesheet.qss"): device not open
Warning: Using function expressions as statements in scripts is not compliant with the ECMAScript specification:
 "function(o) { for (each in o) { print(each); } }..." 
This will throw a syntax error in Qt 5.12. If you want a function expression, surround it by parentheses.
Unknown property link-color
Unknown property link-color
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
ERROR: Unable to claim interface 1:8:5: Device or resource busy (16)
E20241016 11:34:04.782114    11 devicegeneric.cpp:650] Device: Cannot set the number of kernel buffers
```
However, this is all printed before crashing. Nothing new is printed when I click calibrate.

**To Reproduce**
1. Connect ADALM2000
2. Run Scopy
3. Connect to ADALM2000
4. Click calibrate

I am not familiar with this software or the ADALM2000, this is my first time using it. This is a fresh install of Scopy without any changed configuration. I would like to know how to either not experience this or work around it, since I need it for an assignment due tomorrow.

Edit: I forgot to do this. `sudo dmesg` shows:

```
[ 4725.538978] scopy[48497]: segfault at 0 ip 00005864a360f8e3 sp 00007ffe67ff2bd0 error 4 in scopy[39b8e3,5864a330f000+364000] likely on CPU 3 (core 1, socket 0)
[ 4725.538988] Code: 00 00 00 00 0f 1f 00 f3 0f 1e fa 53 48 89 fb 48 83 ec 10 48 8b bf 28 01 00 00 64 48 8b 04 25 28 00 00 00 48 89 44 24 08 31 c0 <48> 8b 07 ff 90 c8 01 00 00 84 c0 0f 85 84 01 00 00 48 8b bb 40 01
[ 4961.587229] scopy[49371]: segfault at 0 ip 0000652ab8c2a8e3 sp 00007ffdf97897c0 error 4 in scopy[39b8e3,652ab892a000+364000] likely on CPU 1 (core 1, socket 0)
[ 4961.587248] Code: 00 00 00 00 0f 1f 00 f3 0f 1e fa 53 48 89 fb 48 83 ec 10 48 8b bf 28 01 00 00 64 48 8b 04 25 28 00 00 00 48 89 44 24 08 31 c0 <48> 8b 07 ff 90 c8 01 00 00 84 c0 0f 85 84 01 00 00 48 8b bb 40 01
[ 5068.653375] scopy[49628]: segfault at 0 ip 0000622526d9f8e3 sp 00007ffd5fbe0d60 error 4 in scopy[39b8e3,622526a9f000+364000] likely on CPU 2 (core 0, socket 0)
[ 5068.653401] Code: 00 00 00 00 0f 1f 00 f3 0f 1e fa 53 48 89 fb 48 83 ec 10 48 8b bf 28 01 00 00 64 48 8b 04 25 28 00 00 00 48 89 44 24 08 31 c0 <48> 8b 07 ff 90 c8 01 00 00 84 c0 0f 85 84 01 00 00 48 8b bb 40 01
```

So this is a segfault.

**Comment by AlexandraTrifan** [comment]
Could you try to update the firmware on your M2K?

**Comment by butterflyhigh** [comment]
> Could you try to update the firmware on your M2K?

I am able to calibrate with the latest firmware, thank you. Could I maybe request adding a note on this issue somewhere in the wiki or README, or possibly somewhere in the code to handle this error instead of segfaulting without explanation? My M2k is very new (purchased in September), so I didn't think of outdated firmware as a potential issue at first. I also figure I'm not the only one who had this issue and spent a long time trying to fix it, and I probably would've fixed it a lot earlier (and finished my assignment earlier LOL) if I had known updating the firmware would resolve it. Once I realized the issue I saw that this is a known issue, but it was very difficult to find information when all I knew was that it segfaulted. 

Either way, I can use Scopy now, I just think it might be nice to have more available info on this. Cheers
